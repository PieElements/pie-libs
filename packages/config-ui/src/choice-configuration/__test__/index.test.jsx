import React from 'react';
import ChoiceConfiguration, { RawChoiceConfiguration } from '../index';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

Enzyme.configure({ adapter: new Adapter() });

const defaultFeedback = {
  correct: 'Correct',
  incorrect: 'Incorrect'
}

const data = {
  checked: true,
  value: 'foo',
  label: 'Foo',
  feedback: {
    type: 'custom'
  }
}

const classes = {
  choiceConfiguration: 'choiceConfiguration'
}

describe('index - snapshot', () => {

  it('renders correctly', () => {

    const tree = renderer
      .create(<ChoiceConfiguration
        defaultFeedback={defaultFeedback}
        data={data}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('index - logic', () => {

  let wrapper, instance, onChange;

  beforeEach(() => {
    onChange = jest.fn();

    wrapper = shallow(<RawChoiceConfiguration
      classes={classes}
      defaultFeedback={defaultFeedback}
      data={data}
      onChange={onChange} />);
    instance = wrapper.instance();
  });

  describe('onValueChange', () => {

    it('calls onChange', () => {

      const event = {
        target: {
          value: 'foo'
        }
      }

      instance.onValueChange(event);
      expect(onChange.mock.calls.length).toBe(1);
      expect(onChange.mock.calls[0][0]).toMatchObject({ value: 'foo' });
    });
  });

  describe('onCheckedChange', () => {
    it('calls onChange', () => {
      instance.onLabelChange('new label');
      expect(onChange.mock.calls.length).toBe(1);
      expect(onChange.mock.calls[0][0]).toMatchObject({ label: 'new label' });
    });
  });


  describe('onFeedbackTypeChange', () => {

    it('calls onChange', () => {
      instance.onFeedbackTypeChange('default');
      expect(onChange.mock.calls[0][0]).toMatchObject({ feedback: { type: 'default' } });
    });
  });

  describe('onFeedbackValueChange', () => {

    it('calls onChange', () => {
      instance.onFeedbackValueChange('new feedback');
      expect(onChange.mock.calls[0][0]).toMatchObject({ feedback: { value: 'new feedback' } });
    });
  });

});
