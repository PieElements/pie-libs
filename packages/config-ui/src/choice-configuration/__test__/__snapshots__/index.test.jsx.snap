// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index - snapshot renders correctly 1`] = `
<div
  className="RawChoiceConfiguration-choiceConfiguration-2"
>
  <div
    className="RawChoiceConfiguration-topRow-3"
  >
    <div
      className="MuiFormControl-root-13 RawInputContainer-formControl-12 RawChoiceConfiguration-toggle-6"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-23 MuiInputLabel-root-17 MuiInputLabel-formControl-18 MuiInputLabel-animated-21 MuiInputLabel-shrink-20"
        data-shrink={true}
      >
        Correct
      </label>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-13 RawChoiceConfiguration-value-4"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-23 MuiInputLabel-root-17 MuiInputLabel-formControl-18 MuiInputLabel-animated-21 MuiInputLabel-shrink-20"
        data-shrink={true}
        htmlFor={undefined}
      >
        Value
      </label>
      <div
        className="MuiInput-root-27 MuiInput-formControl-28 MuiInput-inkbar-29 MuiInput-underline-33"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-36"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value="foo"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-13 RawInputContainer-formControl-12 Component-labelContainer-42"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-23 MuiInputLabel-root-17 MuiInputLabel-formControl-18 MuiInputLabel-animated-21 MuiInputLabel-shrink-20"
        data-shrink={true}
      >
        Label
      </label>
      <div
        className="Component-editorHolder-43"
      >
        <div>
          EditableHtml
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-13 RawInputContainer-formControl-12 RawChoiceConfiguration-feedback-7"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-23 MuiInputLabel-root-17 MuiInputLabel-formControl-18 MuiInputLabel-animated-21 MuiInputLabel-shrink-20"
        data-shrink={true}
      >
        Feedback
      </label>
      <div>
        <div
          onClick={[Function]}
        />
        <div>
          Menu
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-13 RawInputContainer-formControl-12 RawChoiceConfiguration-delete-10"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-23 MuiInputLabel-root-17 MuiInputLabel-formControl-18 MuiInputLabel-animated-21 MuiInputLabel-shrink-20"
        data-shrink={true}
      >
        Delete
      </label>
    </div>
  </div>
  <div
    className="MuiFormControl-root-13 RawInputContainer-formControl-12 Component-labelContainer-42 Component-text-44"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-23 MuiInputLabel-root-17 MuiInputLabel-formControl-18 MuiInputLabel-animated-21 MuiInputLabel-shrink-20"
      data-shrink={true}
    >
      Feedback Text
    </label>
    <div
      className="Component-editorHolder-43"
    >
      <div>
        EditableHtml
      </div>
    </div>
  </div>
</div>
`;
